components/button/README.md

```diff
@@ -1,28 +1,50 @@
 # Buttons
 
-## d2l-button
-
-The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
-
-![Button](./screenshots/button.png?raw=true)
+A Button is used to communicate and perform an action.
 
 ```html
+<!-- docs: demo -->
 <script type="module">
   import '@brightspace-ui/core/components/button/button.js';
+  import '@brightspace-ui/core/components/button/button-subtle.js';
+  import '@brightspace-ui/core/components/button/button-icon.js';
 </script>
-<d2l-button>My Button</d2l-button>
+<d2l-button>Button</d2l-button>
+<d2l-button primary>Primary Button</d2l-button>
+<d2l-button-subtle text="Subtle Button" icon="tier1:gear"></d2l-button-subtle>
+<d2l-button-icon text="Icon Button" icon="tier1:gear"></d2l-button-icon>
 ```
 
-**Properties:**
+## Best Practices
+<!-- docs: start best practices -->
+<!-- docs: start dos -->
+* Use buttons to cause an action or launch a workflow
+* Keep button text short - see "Writing" guidelines
+<!-- docs: end dos -->
+
+<!-- docs: start donts -->
+* Don't use multiple Primary buttons - the primary action should always be clear and obvious
+* Avoid icon-only buttons with unfamiliar icons, only use these for familiar concepts
+* Don't use buttons for navigation, use a link instead
+* Don't open menus with buttons - use a dropdown instead
+<!-- docs: end donts -->
+<!-- docs: end best practices -->
+
+## Button [d2l-button]
+
+The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
 
-| Property | Type | Description |
-|--|--|--|
-| `description` | String | A description to be added to the `button` for accessibility |
-| `disabled` | Boolean | Disables the button |
-| `disabledTooltip` | String | Tooltip text when disabled |
-| `primary` | Boolean | Styles the button as a primary button |
+```html
+<!-- docs: live demo
+name: d2l-button
+-->
+<script type="module">
+  import '@brightspace-ui/core/components/button/button.js';
+</script>
+<d2l-button $attributes>My Button</d2l-button>
```

Comment:
I'm thinking I'll add a hidden-from-docs-site note about ignoring the `$attributes` piece in the demo code.

Reply:
Yeah, I can't seem to find a way that we could "comment" this out so it's not rendered in the readme when it's inside a code block (assuming we could update the the parser to understand that new syntax).

Reply:
Could the docs site make an assumption that the attributes go on the first `<d2l-*>` element inside the code block? That would likely cover 95% of cases and would allow us to avoid the confusion here.

Reply:
I haven't been following this stuff super closely, so I might be a little late to the party here.  If so, and this is already figured out, then ignore.  😏 One concern I would have about assuming the first `d2l-*` element is that some elements require some parent element, so the element of interest would not be the first element.  Consider:
```html
<d2l-list>
	<d2l-list-header slot="header">
		<d2l-selection-action icon="..." text="..." requires-selection></d2l-selection-action>
		<d2l-selection-action icon="..." text="..."></d2l-selection-action>
	</d2l-list-header>
	<d2l-list-item selectable key="..." label="...">
		<img slot="illustration" src="...">
		<d2l-list-item-content>...</d2l-list-item-content>
	</d2l-list-item>
	<d2l-list-item selectable key="..." selected label="...">
		<img slot="illustration" src="...">
		<d2l-list-item-content>...</d2l-list-item-content>
	</d2l-list-item>
	<d2l-list-item selectable key="..." label="...">
		<img slot="illustration" src="...">
		<d2l-list-item-content>...</d2l-list-item-content>
	</d2l-list-item>
</d2l-list>
```
To demo some of these behaviours, these components must be used together... i.e. they can't just be demo'd in isolation.  Do we already have a plan for how to deal with more complex components?

Reply:
Or maybe that could be specified in the `<!-- docs: live demo` snippet?  Something like:
```
<!-- docs: live demo
name: d2l-button
attributes: d2l-button
-->
```
And that's actually a selector, so in Dave B 's case you could use something like:
```
<!-- docs: live demo
name: d2l-list
attributes: d2l-list-item:first-of-type
-->
```
(This might not make sense though with how the parsing/page creation works, not sure at which points we use JS or not)

Reply:
So I think we could likely use what's in `name` for this if we wanted to go this route since `name` corresponds to whatever the component is that's being used for the properties table and then I think we could match that up to the first usage. @GeorgePlukov do you have any thoughts/concerns around this?

Reply:
It could be possible to move this into the demo comment, originally I did it using the `$attributes` for the reason Dave B mentioned where we may have embedded items that we want to be changing the properties on, not just the top level `d2l-*` element. 

Right now the demo code is replacing the $attribute symbol with the attributes as text and updating the playground-project to re-render the demo, so using a css selector wouldn't really work. If we were to use a name property would we just add the attributes to any tag with that name? I was thinking that there could be cases like demoing `d2l-list-item` where we would want a subset of all the items with that tag to update with the attributes

Reply:
I'm a bit on the fence for if it should affect 1 or all. I think all could be useful, since we'd be selective with our demos, e.g., with the `menu-item` specific demo, we might have the first item be a simple menu item and the second be a nested menu, and so having toggling the `disabled` property in the table affect both `d2l-menu-item`s could be neat to see. However this could be complex with the case of `menu-item-radio` where if there were 2 radio items, we wouldn't want `selected` being toggled in the table to affect both. Could also add an option in the comment block for these cases for does it affect all or just first.

Reply:
Yeah thats an interesting case too, Im struggling to think of a better way to go with this so that may be our best option